packages/sanity/src/core/preview/constants.tsx

```diff
@@ -4,6 +4,12 @@ import {type PreviewValue} from '@sanity/types'
 export const INCLUDE_FIELDS_QUERY = ['_id', '_rev', '_type']
 export const INCLUDE_FIELDS = [...INCLUDE_FIELDS_QUERY, '_key']
 
+/**
+ * How long to wait after the last subscriber has unsubscribed before resetting the observable and disconnecting the listener
+ * We want to keep the listener alive for a short while after the last subscriber has unsubscribed to avoid unnecessary reconnects
+ */
+export const LISTENER_RESET_DELAY = 2000
```

Comment:
Is this a sensible value?

Reply:
Let's bump this up to 10 seconds - lazy imports and similar may spend longer than 2s.

Reply:
Good point. Bumped in cb0a4d4e3c